src/forms/form-fields/geometry-field.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import TextField from '../../forms/form-fields/text-field';
+import { isNumber } from 'd2-ui/lib/forms/Validators';
+import { getOr } from 'lodash/fp';
+
+// Allows parsed numbers
+function isNumeric(number) {
+    return !isNaN(parseFloat(number));
+}
+
+function isPoint(poly) {
+    return (
+        Array.isArray(poly) &&
+        (poly.length === 2 && isNumeric(poly[0]) && isNumeric(poly[1]))
+    );
+}
+
+// These actually return true on empty string and undefined, so no error is shown with empty field
+function isValidLatitude(value) {
+    return isNumber(value) && value <= 90 && value >= -90;
+}
+isValidLatitude.message = 'a_latitude_should_be_a_number_between_-90_and_90';
+
+function isValidLongitude(value) {
+    return isNumber(value) && value <= 180 && value >= -180;
+}
+isValidLongitude.message =
+    'a_longitude_should_be_a_number_between_-180_and_180';
+
+function isValidCoordinate(coord) {
+    return isPoint(coord) && isValidLongitude(coord[0]) && isValidLatitude(coord[1])
+}
+const featureTypes = {
+    POINT: 'Point',
+    POLYGON: 'Polygon',
+};
+
+class GeometryField extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {
+            longErr: undefined,
+            latErr: undefined,
+        };
+
+        this.getTranslation = context.d2.i18n.getTranslation.bind(
+            context.d2.i18n
+        );
+    }
+
+    getLongitude = () => getOr('', 'value.coordinates[0]', this.props);
+
+    getLatitude = () => getOr('', 'value.coordinates[1]', this.props);
+
+    getLongLat = () => [this.getLongitude(), this.getLatitude()];
+
+    updateGeometryPoint(long, lat) {
+        const coordinates = [long, lat];
+        const geometry = {
+            type: featureTypes.POINT,
+            coordinates,
+        };
+        if (!long && !lat) {
+            this.props.onChange({ target: { value: null } });
+        } else {
+            this.props.onChange({
+                target: {
+                    value: geometry,
+                },
+            });
+        }
+    }
+
+    handleLatitude = event => {
+        let lat = event.target.value;
+        const long = this.getLongitude();
+        if (!isValidLatitude(lat)) {
```

Comment:
It's nitpicking, but there's an empty line missing here (there's an empty line in `handleLongitude`) for consistency and readability reasons. (Empty line could also be added after the else block)

But this is really just nitpicking..